lxd/identities.go

```diff
@@ -981,6 +989,88 @@ func patchIdentity(d *Daemon, r *http.Request) response.Response {
 	return response.EmptySyncResponse
 }
 
+// swagger:operation DELETE /1.0/auth/identities/tls/{nameOrIdentifier} identities identity_delete_tls
+//
+//	Delete the TLS identity
+//
+//	Removes the TLS identity and revokes trust.
+//
+//	---
+//	produces:
+//	  - application/json
+//	responses:
+//	  "200":
+//	    $ref: "#/responses/EmptySyncResponse"
+//	  "400":
+//	    $ref: "#/responses/BadRequest"
+//	  "403":
+//	    $ref: "#/responses/Forbidden"
+//	  "500":
+//	    $ref: "#/responses/InternalServerError"
+//	  "501":
+//	    $ref: "#/responses/NotImplemented"
+
+// swagger:operation DELETE /1.0/auth/identities/oidc/{nameOrIdentifier} identities identity_delete_oidc
+//
+//	Delete the OIDC identity
+//
+//	Removes the OIDC identity.
+//
+//	---
+//	produces:
+//	  - application/json
+//	responses:
+//	  "200":
+//	    $ref: "#/responses/EmptySyncResponse"
+//	  "400":
+//	    $ref: "#/responses/BadRequest"
+//	  "403":
+//	    $ref: "#/responses/Forbidden"
+//	  "500":
+//	    $ref: "#/responses/InternalServerError"
+//	  "501":
+//	    $ref: "#/responses/NotImplemented"
+func deleteIdentity(d *Daemon, r *http.Request) response.Response {
+	id, err := request.GetCtxValue[*dbCluster.Identity](r.Context(), ctxClusterDBIdentity)
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	if !identity.IsFineGrainedIdentityType(string(id.Type)) {
+		return response.NotImplemented(fmt.Errorf("Identities of type %q cannot be modified via this API", id.Type))
```

Comment:
@tomponline I'm just wondering if this check is really necessary for the delete handler. PUT and PATCH have to be disallowed for non- fine-grained certificates because we don't allow them to be added to groups, but DELETE would work just fine...

Reply:
I'm not sure of the question. Don't we want to ensure that legacy certificates cannot be managed by the identity endpoints?

Reply:
Yeah sorry I'm just having a moment of doubt :laughing: